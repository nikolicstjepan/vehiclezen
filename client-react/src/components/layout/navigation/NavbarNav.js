import React, { Fragment } from "react";
import { Link } from "react-router-dom";

export default function NavbarNav({ isGuest, onLogout }) {
  return (
    <div className="collapse navbar-collapse" id="navbarSupportedContent">
      <ul className="navbar-nav ms-auto mb-2 mb-lg-0">
        {isGuest ? (
          <Fragment>
            <li className="nav-item">
              <Link className="nav-link active" to="/login">
                Login
              </Link>
            </li>
            <li className="nav-item">
              <Link className="nav-link active" to="/register">
                Register
              </Link>
            </li>
          </Fragment>
        ) : (
          <Fragment>
            <li className="nav-item">
              <Link className="nav-link active" to="/">
                Vehicles
              </Link>
            </li>
            <li className="nav-item ms-2">
              <button className="btn btn-outline-light" onClick={onLogout}>
                Logout
              </button>
            </li>
          </Fragment>
        )}
      </ul>
    </div>
  );
}
