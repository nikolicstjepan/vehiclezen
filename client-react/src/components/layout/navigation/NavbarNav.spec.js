import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import NavbarNav from "./NavbarNav";

describe("Navbar", () => {
  test("snapshot renders", () => {
    const component = renderer.create(
      <MemoryRouter>
        <NavbarNav isGuest={true} logout={() => {}} />
      </MemoryRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders login and register when user is guest", () => {
    const component = mount(
      <MemoryRouter>
        <NavbarNav isGuest={true} logout={() => {}} />
      </MemoryRouter>
    );

    expect(component.find("a").at(0).text()).toBe("Login");
    expect(component.find("a").at(1).text()).toBe("Register");
  });

  test("renders logout and vehicles when user is guest", () => {
    const component = mount(
      <MemoryRouter>
        <NavbarNav isGuest={false} logout={() => {}} />
      </MemoryRouter>
    );

    expect(component.find("a").at(0).text()).toBe("Vehicles");
    expect(component.find("button").at(0).text()).toBe("Logout");
  });
});
