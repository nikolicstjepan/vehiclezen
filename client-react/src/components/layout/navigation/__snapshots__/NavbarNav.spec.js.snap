// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot renders 1`] = `
<div
  className="collapse navbar-collapse"
  id="navbarSupportedContent"
>
  <ul
    className="navbar-nav ms-auto mb-2 mb-lg-0"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link active"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </li>
  </ul>
</div>
`;
